packages/hadron-document/src/element.ts

```diff
@@ -67,6 +69,7 @@ export class Element extends EventEmitter {
   invalidTypeMessage?: string;
   decrypted: boolean;
   expanded = false;
+  visibleElementsCount = DEFAULT_VISIBLE_ELEMENTS;
```

Comment:
I'd probably consider a slightly different approach and make this particular property private, and for public would write something that also takes expanded into account, because otherwise you can have a case where expanded is false and reading this value will tell you that there are visible elements, that is not true

Reply:
ahh I think the name led to slight confusion. Maybe this property should be named - maxVisibleElements because that is what it is doing, setting a cap for maximum elements that can be made visible at a time and:
- to actually get the visible elements you use [getVisibleElements](https://github.com/mongodb-js/compass/pull/6009/files/d3418339ee6be5e0fedfe49c160281ecb134ca29#diff-6f9c0f59fcb29fb98792df56908d66e6b4a45b2f5187904e70afdf09053bb52dR842)
- to get a count of visible elements you use [getTotalVisibleElementsCount](https://github.com/mongodb-js/compass/pull/6009/files/d3418339ee6be5e0fedfe49c160281ecb134ca29#diff-6f9c0f59fcb29fb98792df56908d66e6b4a45b2f5187904e70afdf09053bb52dR857)

(I just realised that those methods should account for expanded or not in some way, I will add that.)

Reply:
addressed this in - [dea3187](https://github.com/mongodb-js/compass/pull/6009/commits/dea318758d4165f6a588efbcb36f098181833735)